ecosystem/sep-0023.md

```diff
@@ -0,0 +1,140 @@
+## Preamble
+
+```
+SEP: 0023
+Title: Augmented strkey format for multiplexed addresses
+Author: David Mazières and Tomer Weller
+Track: Standard
+Status: Draft
+Created: 2019-09-16
+Discussion: <link to where discussion for this SEP is taking place, typically the mailing list>
+```
+
+## Simple Summary
+
+Strkey is an ASCII format for representing Stellar account IDs and
+addresses.  This document introduces augmented strkey to represent
+account IDs that are multiplexed across users or separate transactions
+and hence must be used in conjunction with a particular memo ID.
+
+## Motivation
+
+A common pattern in the Stellar ecosystem is for services to share a
+single Stellar account ID across many users or independent
+transactions, relying on the memo ID to disambiguate incoming
+payments.  Unfortunately, experience shows that people frequently
+forget to include the memo ID, resulting in either lost funds or
+onerous support calls.  Hence, we need a way to specify an address and
+memo together in a way that is difficult for end-users to misuse.
+
+## Abstract
+
+A new strkey format is introduced for _multiplexed_ addresses that can
+only be used as a destination address and only in conjunction with a
+particular memo ID.
+
+## Specification
+
+To multiplex a Stellar account over multiple users, the account must
+be used in conjunction with a particular memo ID.  Because the memo
+field is per-transaction rather than per-operation, this places
+restrictions on the use of such multiplexing.  In particular:
+
+* A multiplexed address can occur only as the `destination` field of a
+  `PAYMENT`, `PATH_PAYMENT`, or `ACCOUNT_MERGE` operation.
+
+* At most one multiplexed address can appear in a given transaction.
+
+### Native representation
+
+Because multiplexed addresses cannot be used everyplace a normal
+account can be, implementations must represent them using a distinct
+type.  Since this representation depends on the language, we give a
+suggested implementation in XDR:
+
+~~~ {.c}
+struct MultiplexedAccountID {
+  uint64 *memoID;
```

Comment:
What's the motivation for placing the optional field at the beginning of the struct?